storage/src/test/java/org/apache/kafka/admin/RemoteTopicCrudTest.java

```diff
@@ -0,0 +1,833 @@
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import scala.collection.immutable.Map$;
+import scala.jdk.javaapi.CollectionConverters;
+import scala.jdk.javaapi.OptionConverters;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
+@ClusterTestDefaults(
+        types = Type.KRAFT,
+        brokers = 2,
+        serverProperties = {
+            @ClusterConfigProperty(key = RemoteLogManagerConfig.REMOTE_LOG_STORAGE_SYSTEM_ENABLE_PROP, value = "true"),
+            @ClusterConfigProperty(key = RemoteLogManagerConfig.REMOTE_STORAGE_MANAGER_CLASS_NAME_PROP, value = "org.apache.kafka.server.log.remote.storage.NoOpRemoteStorageManager"),
+            @ClusterConfigProperty(key = RemoteLogManagerConfig.REMOTE_LOG_METADATA_MANAGER_CLASS_NAME_PROP, value = "org.apache.kafka.server.log.remote.storage.NoOpRemoteLogMetadataManager"),
+            @ClusterConfigProperty(key = "log.retention.ms", value = "2000"),
+            @ClusterConfigProperty(key = RemoteLogManagerConfig.LOG_LOCAL_RETENTION_MS_PROP, value = "1000"),
+            @ClusterConfigProperty(key = "retention.bytes", value = "2048"),
+            @ClusterConfigProperty(key = RemoteLogManagerConfig.LOG_LOCAL_RETENTION_BYTES_PROP, value = "1024")
+        }
+)
+
+class RemoteTopicCrudTest {
+
+    private final ClusterInstance cluster;
+    private final int numPartitions = 2;
+    private final int numReplicationFactor = 2;
+
+    private String testTopicName;
+
+    public RemoteTopicCrudTest(ClusterInstance cluster) {
+        this.cluster = cluster;
+    }
+
+    @BeforeEach
+    void setUp(TestInfo info) {
+        var methodName = info.getTestMethod().orElseThrow().getName();
+        testTopicName = methodName + "-" + UUID.randomUUID().toString().replace("-", "").substring(0, 3);
+    }
+
+    @ClusterTest
+    void testCreateRemoteTopicWithValidRetentionTime() {
+        var topicConfig = new Properties();
+        topicConfig.put(TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, "true");
+        topicConfig.put(TopicConfig.RETENTION_MS_CONFIG, "200");
+        topicConfig.put(TopicConfig.LOCAL_LOG_RETENTION_MS_CONFIG, "100");
+        try (var admin = cluster.admin()) {
+            TestUtils.createTopicWithAdmin(admin, testTopicName,
+                    CollectionConverters.asScala(new ArrayList<>(cluster.brokers().values())),
+                    CollectionConverters.asScala(new ArrayList<>(cluster.controllers().values())), 
+                    numPartitions, 
+                    numReplicationFactor,
+                    Map$.MODULE$.empty(),
+                    topicConfig);
+        }
```

Comment:
Please don't use scala TestUtils, we could use `Admin` API 
```suggestion
            admin.createTopics(List.of(new NewTopic(
                    testTopicName,
                    numPartitions,
                    (short) numReplicationFactor).configs(topicConfig))
            );
```

And please update other usage

Reply:
I’ve removed all calls to the Scala `TestUtils` and switched topic-related operations to the `Admin` API.